testsuite/cluster-test/src/experiments/versioning_test.rs

```diff
@@ -0,0 +1,262 @@
+            let instance = all_instances.remove(rnd.gen_range(0, all_instances.len()));
+            instances.push(instance);
+        }
+
+        Self::E {
+            first_batch: instances,
+            second_batch: all_instances,
+            full_nodes: cluster.fullnode_instances().to_vec(),
+            updated_image_tag: self.updated_image_tag,
+        }
+    }
+}
+
+async fn update_batch_instance(
+    context: &mut Context<'_>,
+    updated_instance: &[Instance],
+    updated_tag: String,
+) -> anyhow::Result<()> {
+    let deadline = Instant::now() + Duration::new(2 * 60, 0);
+
+    // 1. Stop Existing instances.
+    let futures: Vec<_> = updated_instance.iter().map(Instance::stop).collect();
+    try_join_all(futures).await?;
+
+    // 2. Reinstantiate a set of new nodes.
+    let futures: Vec<_> = updated_instance
+        .iter()
+        .map(|instance| {
+            let mut newer_config = instance.instance_config().clone();
+            newer_config.replace_tag(updated_tag.clone()).unwrap();
+            context
+                .cluster_swarm
+                .spawn_new_instance(newer_config, false)
+        })
+        .collect();
+    let instances = try_join_all(futures).await?;
+
+    // 3. Wait for the instances to recover.
+    let futures: Vec<_> = instances
+        .iter()
+        .map(|instance| instance.wait_json_rpc(deadline))
+        .collect();
+    try_join_all(futures).await?;
+    Ok(())
+}
+
+#[async_trait]
+impl Experiment for ValidatorVersioning {
+    fn affected_validators(&self) -> HashSet<String> {
+        instance::instancelist_to_set(&self.first_batch)
+            .union(&instance::instancelist_to_set(&self.second_batch))
+            .cloned()
+            .collect()
+    }
+
+    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+        // Mint a number of accounts
+        context
+            .tx_emitter
+            .mint_accounts(
+                &EmitJobRequest::for_instances(
+                    context.cluster.validator_instances().to_vec(),
+                    context.global_emit_job_request,
+                ),
+                150,
+            )
+            .await?;
+
+        info!("1. Changing the images for the instances in the first batch");
+        update_batch_instance(context, &self.first_batch, self.updated_image_tag.clone()).await?;
+
+        info!("2. Send a transaction to make sure it is not rejected nor cause any fork");
+        let full_node = context.cluster.random_full_node_instance();
+        let mut full_node_client = full_node.json_rpc_client();
+        let mut account_1 = context.tx_emitter.take_account();
+        let account_2 = context.tx_emitter.take_account();
+
+        let txn_payload = TransactionPayload::Script(encode_transfer_with_metadata_script(
+            lbr_type_tag(),
+            account_2.address,
+            1,
+            vec![],
+            vec![],
+        ));
+
+        let txn1 = create_user_txn(
```

Comment:
This call (as well as `.sequence_number+=1`) seem to be repeating in this fn - since this is fairly large call, do you think it make sense to make separate function for it?

Reply:
Did you think this can be refactored to avoid code duplication below?